packages/expo-notifications/README.md

```diff
@@ -0,0 +1,504 @@
+2. Channel description (used by android settings).
+3. Notification priority. (range <1,5>)
+4. True if notification should play sound. (false by default);
+5. True - default vibration  (true by default)  
+   False - no vibration  
+   Array of numbers - custom vibration (example: [0, 500, 0, 500]).
+6. True - will increase badge number and False otherwise. (True by default).   
+
+### ActionType
+
+```ts
+type ActionType = {
+  actionId: string;                        //1
+  buttonTitle: string;                     //2
+  isDestructive?: boolean;                 //3
+  isAuthenticationRequired?: boolean;      //4
+  doNotOpenInForeground?: boolean;         //5 
+  textInput?: {
+    submitButtonTitle: string;             //6  
+    placeholder: string;                   //7
+  };
+};
+```
+
+1. Action id.
+2. Title of notification button.
+3. (iOS only) If this property is truthy, on iOS the button title will be highlighted (as if   
+ [this native option](https://developer.apple.com/documentation/usernotifications/unnotificationactionoptions1648199-destructive) was set).
+4. iOS only) If this property is truthy, triggering the action will require authentication from the user (as if [this native option](https://developer.apple.com/documentation/usernotifications/unnotificationactionoptions/1648196-authenticationrequired) was set).
+5. (iOS only) If this property is truthy, triggering the action will not open the app in foreground (as if [this native option](https://developer.apple.com/documentation/usernotifications/unnotificationactionoptions/unnotificationactionoptionforeground) was **NOT** set).
+6. Title of submit title.
+7. Text to display in the text input field.
+
+
+### UserInteraction
+
+```ts
+type UserInteraction = Notification & {
+    actionId?: string;                     //1 
+    userText?: string;                     //2
+    remote?: boolean;                      //3
+}
+```
+
+1. Id of action button that was tapped.  
+2. Text tapped by user.
+3. True if it is push notification.
+
+### OnUserInteractionListener
+
+```ts
+type OnUserInteractionListener = (userInteraction: UserInteraction) => Promise<void>;
+```
+
+### OnForegroundNotificationListener
+
+```ts
+type OnForegroundNotificationListener = (notification: ForegroundNotification) => Promise<void>;
+```
+
+### OnTokenChangeListener
+
+```ts
+type OnTokenChangeListener = (token: string) => Promise<void>;
+```
+
+## Push Notifications
+
+### Expo as a middleman
+
+[How to send Expo push notification](https://docs.expo.io/versions/v35.0.0/guides/push-notifications/#sending-notifications).
+
+### Firebase Push Notifications
+
+Firebase message format suitable for `expo-notifications`.
+
+```json
+{ 
+    "message":{
+        "token":"your_token",
+        "data": {
+            "title": "title",
+            "message": "example content",
+            "channelId": "channelId",
+            "categoryId": "category",
+            "icon": "icon",
```

Comment:
What is `icon`? Is it `path`/`filename`/`url`? Please at least provide hint where to find some clues about this... 😞